sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -247,6 +247,9 @@ internal static bool TryExtractErrorContent(Response response, out ResponseError
         {
             error = null;
             data = null;
+            string? message = null;
+            string? code = null;
```

Comment:
Why are we adding these?  Shouldn't this populate error in either case?

Reply:
I'm not sure I understand the question. These are holder values to populate as we iterate through the json. We then use them to construct the `ResponseError` at the end.

Reply:
> Do you think it's best to have multiple ways to deserialize an error from JSON?

Which other ways are you referring to?

Reply:
The [deserialization code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResponseError.cs#L87) in ResponseError

Reply:
Good point - yes, this may be a good reason to just go with the more basic approach discussed. Updating.